src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -212,52 +202,40 @@ internal static int Encrypt(SafeSslHandle context, byte[] input, int offset, int
                     case Ssl.SslErrorCode.SSL_ERROR_ZERO_RETURN:
                     case Ssl.SslErrorCode.SSL_ERROR_WANT_READ:
                         break;
-
+                    //indicates we need to write the out buffer and write again
+                    case Ssl.SslErrorCode.SSL_ERROR_WANT_WRITE:
+                        break;
                     default:
                         throw new SslException(SR.Format(SR.net_ssl_encrypt_failed, errorCode), innerError);
                 }
             }
-            else
-            {
-                int capacityNeeded = Crypto.BioCtrlPending(context.OutputBio);
-
-                if (output == null || output.Length < capacityNeeded)
-                {
-                    output = new byte[capacityNeeded];
-                }
-
-                retVal = BioRead(context.OutputBio, output, capacityNeeded);
-            }
-
-            return retVal;
+            return context.OutputBio.TakeBytes(out output);
```

Comment:
We are using the existing one. It's always the one we passed into the Bio, we could have a "check if its the same" but then we need to expose the array, and we would need a different signature to above.